share/completions/screen.fish

```diff
@@ -1,18 +1,52 @@
-function __fish_complete_screen --description "Print a list of running screen sessions"
-    screen -list | string match -r '^\t.*\(.*\)\s*$'| string replace -r '\t(.*)\s+\((.*)\)' '$1\tScreen:$2'
+function __fish_detect_screen_socket_dir --description "Detect which folder screen uses"
+    set screen_bin screen
+    if not set -q __fish_screen_socket_dir
+        set -g __fish_screen_socket_dir (eval $screen_bin -ls __fish_i_don_t_think_this_will_be_matched | string match -r "(?<=No Sockets found in ).*(?=\.)")
```

Comment:
Looking over Apple's released screen code, while the per-user TMPDIR thing is an Apple addition, I don't think it's the case that only Macs will have an old screen with no date output (I could be wrong?).

I worry checking for this specific string might break for localizations. Could we try to detect the path whether or not there were attached screens? `-q -list` and exit status might help but probably isn't necessary.